frontend/src/components/object/ObjectProperties.vue

```diff
@@ -1,13 +1,55 @@
 <script setup lang="ts">
+import { onMounted, ref, watch } from 'vue';
+import { storeToRefs } from 'pinia';
+import { useBucketStore, useUserStore } from '@/store';
+import GridRow from '@/components/form/GridRow.vue';
+import { RouteNames } from '@/utils/constants';
 import { formatDateLong } from '@/utils/formatters';
 
-defineProps({
+import type { Ref } from 'vue';
+import type { Bucket } from '@/interfaces';
+
+const bucketStore = useBucketStore();
+const userStore = useUserStore();
+
+const { userSearch } = storeToRefs(userStore);
+
+const props = defineProps({
   objectProperties: {
     type: Object,
     default: undefined
+  },
+  fullView: {
+    type: Boolean,
+    default: true
   }
 });
 
+const bucket: Ref<Bucket | undefined > = ref();
+const createdBy: Ref<string | undefined> = ref();
+const updatedBy: Ref<string | undefined> = ref();
+
+async function getBucketData() {
+  // Get some associated bucket information
+  bucket.value = await bucketStore.getBucketInfo(props.objectProperties?.bucketId);
+}
+
+async function getUserData() {
+  await userStore.searchUsers({userId:[props.objectProperties?.createdBy, props.objectProperties?.updatedBy]});
+  createdBy.value = userSearch.value.find( x => x.userId === props.objectProperties?.createdBy )?.fullName;
+  updatedBy.value = userSearch.value.find( x => x.userId === props.objectProperties?.updatedBy )?.fullName;
+}
+
+onMounted(() => {
+  getBucketData();
+  getUserData();
+});
+
+watch( props, async () => {
+  await getBucketData();
+  await getUserData();
+});
```

Comment:
Async/await: Is there an actual linear dependency where getUserData needs to wait on getBucketData to complete first? If not, we should let these both fire asynchronously. That means the following is sufficient:
```ts
watch(props, () => {
  getBucketData();
  getUserData();
});
```

Reply:
What's the need for the watches in these components. It's a load->loading->display pattern right? Or are there refreshes somewhere or subscriptions that would update that data?

Reply:
For the sidebar in the object list, the first time it is opened it gets mounted but if you click the info icon again on a different object it doesn't remount, it just receives new props.